CMakeLists.txt

```
@@ -0,0 +1,198 @@
+cmake_minimum_required(VERSION 3.14)
+project(mold VERSION 1.3.0)
+
+include(CMakeDependentOption)
+
+# FIXME: this is for parity with the makefiles that install directly to <prefix>/lib
+# NOTE: defining this before including GNUInstallDirs makes its platform detection a noop
+set(CMAKE_INSTALL_LIBDIR lib CACHE PATH "")
+include(GNUInstallDirs)
+
+# Mold build options
+cmake_dependent_option(MOLD_USE_MIMALLOC "Use mimalloc" ON
+    "NOT APPLE;NOT ANDROID" OFF)
+cmake_dependent_option(MOLD_USE_SYSTEM_MIMALLOC "Use system or vendored mimalloc" ON
+    MOLD_USE_MIMALLOC OFF)
+option(MOLD_USE_SYSTEM_TBB "Use system or vendored TBB" ON)
+
+# Setup mimalloc
+if(MOLD_USE_MIMALLOC)
+    if(MOLD_USE_SYSTEM_MIMALLOC)
+        find_package(mimalloc REQUIRED)
+        target_compile_definitions(mimalloc INTERFACE USE_SYSTEM_MIMALLOC)
```

I forgot you can't modify IMPORTED targets like this.

```suggestion
        set_property(TARGET mimalloc APPEND PROPERTY INTERFACE_COMPILE_DEFINITIONS USE_SYSTEM_MIMALLOC)
```

---

Shouldn't this be equivalent? Also I just checked and `USE_SYSTEM_MIMALLOC` gets defined when using `target_compile_definitions`.